jax/scipy/optimize/bfgs_minimize.py

```diff
@@ -0,0 +1,128 @@
+"""The Broyden-Fletcher-Goldfarb-Shanno minimization algorithm.
+https://pages.mtu.edu/~struther/Courses/OLD/Sp2013/5630/Jorge_Nocedal_Numerical_optimization_267490.pdf
+"""
+
+import jax
+import jax.numpy as jnp
+from jax.lax import while_loop
+from .line_search import line_search
+from typing import NamedTuple, Optional
+
+BFGSResults = NamedTuple(
+    'BFGSResults', [
+        ('converged', bool),  # bool, True if minimization converges
+        ('failed', bool),  # bool, True if line search fails
+        ('k', int),  # The number of iterations of the BFGS update.
+        ('nfev', int),  # The total number of objective evaluations performed.
+        ('ngev', int),  # total number of jacobian evaluations
+        ('nhev', int),  # total number of hessian evaluations
+        ('x_k', jnp.ndarray),
+        # A tensor containing the last argument value found during the search. If the search converged, then
+        # this value is the argmin of the objective function.
+        ('f_k', jnp.ndarray),  # A tensor containing the value of the objective
+        # function at the `position`. If the search
+        # converged, then this is the (local) minimum of
+        # the objective function.
+        ('g_k', jnp.ndarray),  # A tensor containing the gradient of the
+        # objective function at the
+        # `final_position`. If the search converged
+        # the max-norm of this tensor should be
+        # below the tolerance.
+        ('H_k', jnp.ndarray)  # A tensor containing the inverse of the estimated Hessian.
+    ])
+
+
+def bfgs_minimize(func, x0, options=None):
+    """
+        Minimisation with BFGS. Inexact line search is performed satisfying both Wolfe conditions
```

Comment:
This docstring has one more tab of indentation than needed, e.g., per the example here:
http://google.github.io/styleguide/pyguide.html#doc-function-raises

Also: we use two space indentations in JAX currently. Please adjust your code to match.